migrator/tests/actions/drafts/test_draft_actions_stream.py

```diff
@@ -24,25 +24,10 @@
 
 from zenodo_rdm_migrator.transform.transactions import ZenodoTxTransform
 
-
-@pytest.fixture()
-def test_extract_cls(create_draft_tx):
-    class TestExtractor(Extract):
-        """Test extractor."""
-
-        def run(self):
-            """Yield one element at a time."""
-            yield Tx(
-                id=create_draft_tx["tx_id"], operations=create_draft_tx["operations"]
-            )
-
-    return TestExtractor
-
-
 DB_URI = "postgresql+psycopg://invenio:invenio@localhost:5432/invenio"
 
 
-@pytest.fixture(scope="function")
+@pytest.fixture(scope="module")
 def db_engine():
```

Comment:
minor: maybe it's worth to borrow the fixture we have in `pytest-invenio` for the DB, which creates a transaction and rolls it back as clean-up? This might improve the speed issues we've discussed (and is a best-practice/recipe described in SQLAlchemy docs too)

Reply:
working on this atm

Reply:
I'm quitting on this, it is very time consuming to figure it out and I dont think we will gain much in performance (it would make some test cleaner since they have atm some nasty teardown/cleanup code.

Main issues:
- It is not just a copy past from pytest-invenio
- Pytest-invenio works with flask database object and sqlalchemy 1.4
- This module works with the Engine API (not database) and SQLAlchemy 2
- I was able to make a fixture that rollsback operations executed on it, however, when we run test with `action.run()` there is a new engine and session created internally and I did not manage to make the event listener work with the engine API.